File: Euclid/Messages/Incoming/INFORETRIEVE.cs
using Euclid.Game;
using Euclid.Messages.Outgoing;
using Euclid.Network.Streams;
using Euclid.Storage.Database.Access;

namespace Euclid.Messages.Incoming
{
    class INFORETRIEVE : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            string username = request.GetArgument(0);
            string password = request.GetArgument(1);

            var playerData = UserDao.Login(username, password);

            if (playerData == null)
            {
                return;
            }

            player.Send(new USEROBJECT(playerData));
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/INFORETRIEVE.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.messages.IMessageEvent;
import euclid.network.streams.Request;
import euclid.storage.database.access.UserDao;
import euclid.messages.outgoing.USEROBJECT;
import euclid.storage.database.data.PlayerData;

public class INFORETRIEVE implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        String username = request.getArgument(0);
        String password = request.getArgument(1);

        PlayerData playerData = UserDao.login(username, password);

        if (playerData == null) {
            return;
        }

        player.send(new USEROBJECT(playerData));
    }
}
